### DISCUSSION TITLE
LTO/WPO and other optimizations

### DISCUSSION THREAD

**Discussion by petersasi** [petersasi/deluge2-win-build#27] [General]
Hi @mhertz,
Let me summarize my views on this:

1. I started with libtorrent because I had already changed the optimize for speed there and there is quite some hasing there, imagine that for the thousands of torrents, millions of pieces use case. Luckily the existing B2 option made this very easy for 1.2RC
2. I have not checked if libtorrent 1.1 and 1.2.3 can be built with a recent enough boost already having the LTO option.
3. Great thanks to for already delivering the OpenSSL build with LTO.
4. Another candidate that may use it is the GTK (GVSbuild) to make the UI faster too.

As for the concerns: 

- several linux distros started using it for most of their packages, so I expect it not to decrease performance
- size: I do not care much for slightly larger executables, but LTO theoretically could help reducing too (I am unsure MSVC does dead code elimination, GCC and LLVM does)

**Comment by mhertz** [comment]
Thanks Peter, appreciated :) Yeah, I said some wrong things previously, e.g. found later several of the bigger distros are in process of changing or have, or have a timeline kinda thing with it. Anyway I know diddly-squat about optimizations and whatnot, so nice that you're on-top of those things and forefronted them :) I was little hessitant at first, not because of size in general but if was an indicator maybe of that it haden't worked as aspected maybe, but have read little up on it and seen many state it goes little up sometimes, and that the size reduction ussually seen, just is nice side-point, but as seen I added it in myself in all libtorrents and openssl, so no concerns anymore :) I checked the older libtorrents with filedump adding flags in the way I did, did work, as both size went little up and filedump states linked with /ltcg etc, and also the /lto flags of newer boosts does the same as I did anyway, I saw from the boost code afterwards.

Off-topic, but I cleaned up little and deleted my older tickets in your issues section, as thought no reason to clutter with my longwinded thoughts of stuff, when already read once, and no actual issues.
